platform/iphone/os_iphone.cpp

```diff
@@ -356,14 +356,32 @@ void OSIPhone::delete_main_loop() {
 
 void OSIPhone::finalize() {
 
-	if (main_loop) // should not happen?
-		memdelete(main_loop);
+	delete_main_loop();
```

Comment:
`delete_main_loop()` is already called in `Main::cleanup()`, so it shouldn't be necessary here. That's also why the comment meant, though it points that maybe something is not working properly if `main_loop` is not `NULL` here.

Reply:
I was confused why that snippet and comment were there so I compared the `OSIPhone::finalize()` function to the `OS_OSX::finalize()` function for the MacOSX platform and there its `delete_main_loop()` is called inside the  `OS_OSX::finalize()` function.

https://github.com/zxcvdev/godot/blob/9a90ea14259edc6596fa73e03d635ca822d9ac37/platform/osx/os_osx.mm#L1656-L1674

So inside `OSIPhone::finalize()` I replaced the mentioned snippet with `delete_main_loop()` to be uniform since the content of `delete_main_loop()` is functionally the same as the replaced code.

https://github.com/zxcvdev/godot/blob/9a90ea14259edc6596fa73e03d635ca822d9ac37/platform/iphone/os_iphone.cpp#L347-L355

But if you want me to leave it as it was, let me know.

Reply:
I think it's fine to use this as you did, especially if macOS also does it, which confirms that the `Main::cleanup()` is not sufficient (or is not called on those platforms maybe?).

Needs a good refactoring pass, but for the scope of fixing the crash for 3.2 that's definitely good as is.